projects/plugins/jetpack/extensions/blocks/ai-assistant/edit.js

```diff
@@ -0,0 +1,444 @@
+import './editor.scss';
+
+import { useAnalytics } from '@automattic/jetpack-shared-extension-utils';
+import apiFetch from '@wordpress/api-fetch';
+import { useBlockProps, store as blockEditorStore } from '@wordpress/block-editor';
+import { rawHandler, createBlock } from '@wordpress/blocks';
+import {
+	Button,
+	DropdownMenu,
+	Flex,
+	FlexBlock,
+	Modal,
+	TextareaControl,
+	// eslint-disable-next-line wpcalypso/no-unsafe-wp-apis
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	// eslint-disable-next-line wpcalypso/no-unsafe-wp-apis
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+} from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { Icon, pencil, moreVertical } from '@wordpress/icons';
+import MarkdownIt from 'markdown-it';
+import ImageWithSelect from './image-with-select';
+import { getImagesFromOpenAI } from './lib';
+import Loading from './loading';
+import ShowLittleByLittle from './show-little-by-little';
+
+// Maximum number of characters we send from the content
+export const MAXIMUM_NUMBER_OF_CHARACTERS_SENT_FROM_CONTENT = 1024;
+
+// Creates the prompt that will eventually be sent to OpenAI. It uses the current post title, content (before the actual AI block) - or a slice of it if too long, and tags + categories names
+export const createPrompt = (
+	postTitle = '',
+	contentBeforeCurrentBlock = [],
+	// eslint-disable-next-line no-unused-vars
+	categoriesNames = '',
+	// eslint-disable-next-line no-unused-vars
+	tagsNames = '',
+	userPrompt = '',
+	type = 'userPrompt'
+) => {
+	const promptSuffix = __(
+		'. Please always output the generated content in markdown format. Do not include a top level heading by default. Please only output generated content ready for publishing.',
```

Comment:
Is it ok defining the prompt here? I'm not saying it's a blocker but should we build it on the server side? 
Just FYI, this line in some way is triggering the failing test:

https://github.com/Automattic/jetpack/actions/runs/4917595635/jobs/8782896803?pr=30444#step:9:501

```
expect(received).toBeFalsy()
  [test-extensions] 
      Received: ". Please always output the generated content in markdown format. Do not include a top level heading by default. Please only output generated content ready for publishing."
```

Reply:
It could be built server-side eventually indeed. There's [an issue](https://github.com/orgs/Automattic/projects/667/views/1?pane=issue&itemId=27456467) about it in [the board](https://github.com/orgs/Automattic/projects/667/views/1)

Reply:
Just a side note, I like how polite it is.

Reply:
Trying portuguese prompts here and the language mismatch between the prompt and the suffix is not a problem, so it does not need to be translated as well, another reason to do it server-side.